backend/src/statistics/tests/statistics.e2e.spec.ts

```diff
@@ -0,0 +1,193 @@
+import { INestApplication } from "@nestjs/common";
+import { Test } from "@nestjs/testing";
+import { Language } from "@prisma/client";
+import { LoggerModule } from "nestjs-pino";
+import request from "supertest";
+
+import { AuthModule } from "src/auth/auth.module";
+import { MockSessionMiddleware } from "src/common/middlewares/test/MockSessionMiddleware";
+import { PrismaServiceModule } from "src/core/prisma.service.module";
+import { TestClient } from "src/core/test/test-client";
+import { RedisServiceModule } from "src/redis/redis.service.module";
+
+import { StatisticsModule } from "../statistics.module";
+
+import { Difficulty } from "~shared/types/base";
+import { Status } from "~shared/types/base";
+
+describe("User", () => {
+  let app: INestApplication;
+  const client: TestClient = new TestClient();
+
+  beforeAll(async () => {
+    const module = await Test.createTestingModule({
+      imports: [
+        // Module under test
+        StatisticsModule,
+        // Global modules
+        PrismaServiceModule,
+        RedisServiceModule,
+        LoggerModule.forRoot({
+          pinoHttp: {
+            level: "silent",
+          },
+        }),
+        // Dependencies
+        AuthModule,
+      ],
+    }).compile();
+
+    app = module.createNestApplication();
+    // Mock session data.
+    const mockSessionMiddleware = new MockSessionMiddleware();
+    app.use(mockSessionMiddleware.use.bind(mockSessionMiddleware));
+    await app.init();
+  });
+
+  beforeEach(async () => {
+    await client.reset();
+    await client.category.create({
+      data: {
+        id: 1,
+        title: "Algorithms",
+      },
+    });
+    await client.topic.createMany({
+      data: [
+        {
+          id: 1,
+          name: "Dynamic Programming",
+        },
+        {
+          id: 2,
+          name: "Bit Manipulation",
+        },
+        {
+          id: 3,
+          name: "Greedy",
+        },
+        {
+          id: 4,
+          name: "Memoization",
+        },
+      ],
+    });
+    await client.question.create({
```

Comment:
Could we use at least two questions instead to make sure the topics graph is calculated correctly?